timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/HeldLocksCollection.java

```diff
@@ -68,23 +83,34 @@ public void failAllOutstandingRequestsWithNotCurrentLeaderException() {
         heldLocksById.values().forEach(result -> result.failIfNotCompleted(ex));
     }
 
+    private Leased<LockToken> createLeasableLockToken(HeldLocks heldLocks) {
+        return Leased.of(heldLocks.getToken(), leaseWithStart(heldLocks.lastRefreshTime()));
+    }
+
+    private Lease leaseWithStart(NanoTime startTime) {
+        return Lease.of(LeaderTime.of(leaderClock.id(), startTime),
+                LockLeaseContract.CLIENT_LEASE_TIMEOUT);
+    }
+
     private boolean shouldRemove(AsyncResult<HeldLocks> lockResult) {
         return lockResult.isFailed()
                 || lockResult.isTimedOut()
                 || lockResult.test(HeldLocks::unlockIfExpired);
     }
 
-    private Set<LockToken> filter(Set<LockToken> tokens, Predicate<HeldLocks> predicate) {
+    private Leased<Set<LockToken>> filter(Set<LockToken> tokens, Predicate<HeldLocks> predicate) {
```

Comment:
given this is also used in the unlock method, it's probably wrong to return leased here. I think this should just return `Set<LockToken>` and we can wrap with the lease in refresh.